src/Data/Fin/Properties.agda

```diff
@@ -761,6 +767,24 @@ pinch-mono-≤ 0F      {suc j} {suc k} (s≤s j≤k) = j≤k
 pinch-mono-≤ (suc i) {0F}    {k}     0≤n       = z≤n
 pinch-mono-≤ (suc i) {suc j} {suc k} (s≤s j≤k) = s≤s (pinch-mono-≤ i j≤k)
 
+pinch-injective :
+  {k : ℕ} (x : Fin k) {y z : Fin (ℕ.suc k)} →
```

Comment:
because of how Agda add implicits, it's usually a bad idea to alternate explicits and implicits, especially when they are independent like this.  Best to put `x` last.

Reply:
I realize that `x` can also be implicit here. Would it be ok if I make it implicit, or do you prefer to keep it explicit and making it the last one?

Reply:
If they are all implicit, they can stay in the same order as they are in now.

Side comment: I'm always uncomfortable with proofs by induction on implicits. It always makes me feel like I'm doing something wrong, like maybe there's some intermediate lemmas I'm missing or something. I'm not asking you to do anything with this, just something to ponder.

Reply:
There are some cases where agda can't infer the implicit argument if a construction was made by induction on that implicit argument.

Reply:
Right - I've evolved my code to never do constructions by recursion on implicits, that's just asking for trouble. Proofs are another matter.